File: bubble_sort.py
from random import shuffle

l = list(range(1, 1000 + 1))
shuffle(l)
print(f"List before sorting {l} \n")
for i in range(1, len(l)):
    for j in range(len(l) - i):
        if l[j] > l[j + 1]:
            l[j], l[j + 1] = l[j + 1], l[j]
        else:
            continue

print(f"After bubble_sort {l}")

File: bubble_sort.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	someList := getIntRange(0, 10)
	shuffle(someList)
	fmt.Println("Before bubble sort:\n", someList)
	bubbleSort(someList)
	fmt.Println("After bubble sort:\n", someList)
}

func bubbleSort(s []int) []int {
	for i := len(s); i > 0; i-- {
		for j := 1; j < i; j++ {
			if s[j-1] > s[j] {
				s[j-1], s[j] = s[j], s[j-1]
			}
		}
	}
	return s
}

func getIntRange(from, to int) []int {
	sliceRange := make([]int, to-from)
	for i := range sliceRange {
		sliceRange[i] = i + 1
	}
	return sliceRange
}

func shuffle(slice []int) []int {
	rand.Seed(time.Now().UnixNano())
	rand.Shuffle(len(slice), func(i, j int) { slice[i], slice[j] = slice[j], slice[i] })
	return slice
}
